WcaOnRails/app/views/competitions/_nav.html.erb

```diff
@@ -135,20 +135,20 @@
       <h3><%= @competition.name %></h3>
       <hr />
 
-      <% if !@competition.showAtAll? %>
-        <div class="alert alert-warning">
-          <strong>Note:</strong> This competition is not visible to the public.
-        </div>
-      <% end %>
-      <% if !@competition.results_uploaded? %>
-        <% if @competition.is_over? %>
-          <div class="alert alert-info">
-            <strong>Note:</strong> This competition is over, we are working to upload the results as soon as possible!
+      <% if !@competition.warnings.empty? %>
+        <% @competition.warnings.each do |field, message| %>
+          <div class="alert alert-warning">
+            <strong>Note:</strong>
+            <%= message %>
           </div>
         <% end %>
-        <% if @competition.in_progress? %>
+      <% end %>
+
+      <% if !@competition.info.empty? %>
```

Comment:
More rubistic way would be to write `unless` instead of `if !something` but maybe that depends on personal feeling, so it's your choice =)